### DISCUSSION TITLE
How to sum value in specific attributes of multiple specific nodes?

### DISCUSSION THREAD

**Discussion by fedor5792** [freeplane/freeplane#790] [Help] [answered]
Hi! Could you help me? I can find specific nodes and sum their child nodes using "parent.parent.parent.find{(it.text == node.text && it.alias=='Output')}.children.to.num.sum(0)"
I want to find specific nodes and summarize the values of specific attributes, I used "parent.parent.parent.find{(it.text == node.text && it.alias=='Output')}.attributes.findAll { it.key =='Value'}.value.sum(0)"   But it doesn't work.

(If don't use "find" that will work only one node - "parent.parent.parent.attributes.findAll { it.key =='Value'}.value.sum(0)")
[New Mindmap.zip](https://github.com/freeplane/freeplane/files/9900019/New.Mindmap.zip)

**Comment by macmarrum** [comment, marked as answer]
You can use `collect`
```groovy
parent.parent.parent.find{(it.text == node.text && it.alias=='Output')}.collect { it.attributes.findAll { it.key =='Value'}.value }.sum(0)
```
Please see also
- #763

**Reply by fedor5792** [reply]
It works. Thank you!

**Reply by fedor5792** [reply]
I am so sorry. I have a problem. 
"parent.parent.parent.find{(it.text == node.text && it.alias=='Output')}.collect { it.attributes.findAll { it.key =='Value'}.value }.sum(0)"     // value [3], [2].sum(0) =5 ok

But if add a formul

// value [3], [=1+1].sum(0) = Error setting the text for the following input ... No signature of method: java.math BigDecimal.plus() is aplicable for argument types: (ArrayList) value:[[=1+1]].....

".value" get string [=1+1]  i don't know how to fix it, Could you help me please.

**Reply by macmarrum** [reply]
If there is only one attribute with the key `Value`, you can use `it['Value']` instead of `it.attributes.findAll { it.key =='Value' }.value`. The former will find only the first one, the latter will find all of them.

Additionally, `it['Value']` evaluates formulas and allows to treat the value as text, number or date. And `it['Value'].num0` returns 0 when the value is not a number.

The resulting formula will be
```groovy
=parent.parent.parent.find{(it.text == node.text && it.alias=='Output')}.collect { it['Value'].num0 }.sum(0)
```

**Reply by fedor5792** [reply]
Yes, one attribute is enough. I’m really grateful for your help!
